MainApp/lib/app/routes/api.js

```diff
@@ -52,30 +52,35 @@ export default function (router) {
         gameConfig.NHnumOfSurgeons = req.body.NHnumOfSurgeons;
         gameConfig.patientHelpTimeInSeconds = req.body.patientHelpTimeInSeconds;
 
-        if (gameConfig.cooperation == null || gameConfig.cooperation === '' ||
-            gameConfig.mode == null || gameConfig.mode === '' ||
-            gameConfig.earlyType == null || gameConfig.earlyType === '' ||
-            gameConfig.startNumPatientAs == null || gameConfig.startNumPatientAs === '' ||
-            gameConfig.startNumPatientBs == null || gameConfig.startNumPatientBs === '' ||
-            gameConfig.numOfDoctors == null || gameConfig.numOfDoctors === '' ||
-            gameConfig.numOfNurses == null || gameConfig.numOfNurses === '' ||
-            gameConfig.numOfSurgeons == null || gameConfig.numOfSurgeons === '' ||
-            gameConfig.totalTimeInSeconds == null || gameConfig.totalTimeInSeconds === '' ||
-            gameConfig.NHstartNumPatientAs == null || gameConfig.NHstartNumPatientAs === '' ||
-            gameConfig.NHstartNumPatientBs == null || gameConfig.NHstartNumPatientBs === '' ||
-            gameConfig.NHnumOfDoctors == null || gameConfig.NHnumOfDoctors === '' ||
-            gameConfig.NHnumOfNurses == null || gameConfig.NHnumOfNurses === '' ||
-            gameConfig.NHnumOfSurgeons == null || gameConfig.NHnumOfSurgeons === '' ||
-            gameConfig.patientHelpTimeInSeconds == null || gameConfig.patientHelpTimeInSeconds === '') {
-            res.send({success: false, message: 'One or more fields were empty'});
-        }
-        else {
-            gameConfig.save(function (error) {
+        gameConfig.save(function (error) {
+            if (error) {
+                console.log(error);
+                res.send({success: false, message: error.errors});
+            } else {
+                res.send({success: true, message: "Game config saved"});
+            }
+        });
+
+    });
+
+    //http://localhost:8080/api/gameinfo
+    router.post('/gameinfo', function (req, res) {
+        let gameinfo = new Game();
+        gameinfo.gameConfigId = req.body.gameConfigId;
+        gameinfo.trialInfoId = req.body.trialInfoId;
+        gameinfo.userStatsId = req.body.userStatsId;
+        gameinfo.username = req.body.username;
+
+        if (gameinfo.gameConfigId == null || gameinfo.gameConfigId == '' || gameinfo.trialInfoId == null || gameinfo.trialInfoId == '' || gameinfo.userStatsId == null || gameinfo.userStatsId == '') {
```

Comment:
Yeah we should change all those. Apparently the mongoose-validator library is kind of useless. 
Here's a [good tutorial](http://thecodebarbarian.com/2015/02/20/better-angularjs-form-validation-with-mongoose) I found for validation.
I think Tharun is handling this schema.